### PR TITLE
Added mdcheckr to detect broken links in Markdown 

### PR DESCRIPTION
**Pull request by korken89**
This will be the PR for checking for broken links in documentation, I will start adding it to more places as soon as we are happy with it here.
Please come with comments if there is something more that you'd like to add or change.

### CODE REVIEW

File: ci/script.sh [hunk 1]
```diff
@@ -76,6 +76,9 @@ main() {
 
     if [ $TARGET = x86_64-unknown-linux-gnu ]; then
         ./check-blobs.sh
+
+        # Check Markdown files for broken links
+        find . -name '*.md' -print0 | xargs -0 mdcheckr
```
Comment (japaric) [inline ci/script.sh]: is this the same as `mdcheckr **/*.md`? If yes, I find the glob version more readable.

**Comment by korken89** [comment]
Any comments on the CHANGELOG link issue @rust-embedded/cortex-m ? I am unable to get around the scenario @japaric mentioned, unless removing the file from being checked. But it sub-optimal.

**Comment by korken89** [comment]
So eh, should we close this?

**Comment by therealprof** [comment]
@korken89 Sounds good to me.

**Comment by korken89** [comment]
If we want to look into this again we can redo it

**Pull request closed without merging**